sycl/test-e2e/DeviceCodeSplit/grf.cpp

```diff
@@ -22,17 +21,18 @@
 // RUN: %{build} -o %t.out
 // RUN: env SYCL_PI_TRACE=-1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-NO-VAR
 // RUN: env SYCL_PROGRAM_COMPILE_OPTIONS="-g" SYCL_PI_TRACE=-1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-WITH-VAR
-// RUN: %{build} -DUSE_LARGE_GRF=1 -o %t.out
-// RUN: env SYCL_PI_TRACE=-1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-NO-VAR
-// RUN: env SYCL_PROGRAM_COMPILE_OPTIONS="-g" SYCL_PI_TRACE=-1 %{run} %t.out 2>&1  | FileCheck %s --check-prefixes=CHECK,CHECK-WITH-VAR
-
+// RUN: %{build} -DUSE_AUTO_GRF=1 -o %t.out
+// RUN: env SYCL_PI_TRACE=-1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-AUTO-NO-VAR
+// RUN: env SYCL_PROGRAM_COMPILE_OPTIONS="-g" SYCL_PI_TRACE=-1 %{run} %t.out 2>&1 | FileCheck %s --check-prefixes=CHECK,CHECK-AUTO-WITH-VAR
 #include "../helpers.hpp"
 #include <iostream>
-#include <sycl/ext/intel/experimental/kernel_properties.hpp>
+#include <sycl/detail/kernel_properties.hpp>
 #include <sycl/sycl.hpp>
 
 using namespace sycl;
+using namespace sycl::detail;
```

Comment:
sycl::detail::register_alloc_mode ?!

Sorry for the late comment, but I just did not realize this issue during code-review phase and discovered it only now when tried to use the new API.

I believe we must never inspire users to use 'detail' namespace, and ask include files from "\*/detail/\*" - it is only for SYCL internal needs. Only SYCL implementation should use detail namespace, not user.

The enum must have been defined not as "sycl::detail::register_alloc_mode", but as "sycl::ext::oneapi::experimental::register_alloc_mode".

The file kernel_properties.hpp also needed to be placed at ext/oneapi/experimental/kernel_properties.hpp.

Reply:
Maybe the file needs to be placed in ext/intel/experimental/kernel_properties.cpp (where it was), which seems more correct than where it is now (sycl/detail).

Reply:
The detail namespace was picked after the discussion in https://github.com/intel/llvm/pull/9258#discussion_r1195237389 as it was believed to be an internal-only property. If it is meant to be used externally we would normally require documentation of the feature prior to implementation. When we have that, I see no problem in moving it out of the detail namespace.

Reply:
The previous "experimental" extension should not have been removed while there is no new equivalent "experimental" feature. This means that we need to switch from the old "experimental" to new "detail" (code breaks), then we need to switch from "detail" to "experimental" (another code break), and then finally from "experimental" to final (3rd code break). Is there any reason apart from the lack of documentation to add this to "detail" first?